### PR TITLE
To re-register system after media upgrade at system_prepare

### PR DESCRIPTION
**Pull request by tinawang123**
To re-register system after media upgrade, so we can run regression test. As after media upgrade, the system is de-register, but regression test cases need run on registration system.

- Related ticket: https://progress.opensuse.org/issues/60872
- Verification run: 
For sles media upgrade: https://openqa.nue.suse.com/tests/3860246
For sled media upgrade: https://openqa.nue.suse.com/tests/3863001#step/system_prepare/28

### CODE REVIEW

File: tests/console/system_prepare.pm [hunk 1]
```diff
@@ -44,12 +44,18 @@ sub run {
     }
 
     # Register the modules after media migration, so it can do regession
-    if (get_var('SCC_ADDONS') && get_var('MEDIA_UPGRADE') && get_var('KEEP_REGISTERED')) {
+    if (get_var('SCC_ADDONS') && get_var('MEDIA_UPGRADE') && get_var('REGISTER_AFMIG')) {
```
Comment (lemon-suse) [inline tests/console/system_prepare.pm]: Maybe we can set the variable like 'RE_REGISTER_AFTER_UPGRADE', WDYT?

Comment (tinawang123) [reply tests/console/system_prepare.pm]: A little long. Change to REREG_AF_UPG OK?   Don't use too long name

**Comment by lemon-suse** [comment]
I do think you'd better run other test as verify, such as SCC/ProxySCC.

**Comment by tinawang123** [comment]
> I do think you'd better run other test as verify, such as SCC/ProxySCC.

I add the condition at here: get_var('MEDIA_UPGRADE')    If don't set this one, it cannot run those code. So, I don't think it is necessary to run scc/pscc.  Especially,  It also need 'REREG_AF_UPG' this setting. Many condition to need to run this code.

File: tests/console/system_prepare.pm [hunk 2]
```diff
@@ -44,12 +44,18 @@ sub run {
     }
 
     # Register the modules after media migration, so it can do regession
-    if (get_var('SCC_ADDONS') && get_var('MEDIA_UPGRADE') && get_var('KEEP_REGISTERED')) {
+    if (get_var('SCC_ADDONS') && get_var('MEDIA_UPGRADE') && get_var('REREG_AF_UPG')) {
```
Comment (coolgw) [inline tests/console/system_prepare.pm]: REREG_AF_UPG seems not good in my view, not block issue.

**Comment by coolgw** [comment]
if we change code, should we also change the comments in the beginning of the file? i also suggest owner have a look.

File: tests/console/system_prepare.pm (hunk 2 above)
Comment (tinawang123) [reply tests/console/system_prepare.pm]: Any suggestion?

> REREG_AF_UPG seems not good in my view, not block issue.

Any suggestion?

**Comment by okurz** [comment]
Hi @tinawang123 , typo in PR subject line :) Please rebase to exclude the merge commit and change the git commit message to something more detailed than "Change …". The commit message could have a bit more details, e.g. what issue is fixed, why this needs to change, to which versions of which product it applies, link to a bug or a fate entry, the choices you made, etc. Also see https://commit.style/ or http://chris.beams.io/posts/git-commit/ as a helpful guide how to write good commit messages.

**Comment by tinawang123** [comment]
Change condition to run the code 
Verified: https://openqa.nue.suse.com/tests/3873038#

**Comment by coolgw** [comment]
LGTM
Just additional notes, this logic is only for migration regression check.

**Pull request merged**